pyproject.toml

```diff
@@ -21,36 +21,36 @@ classifiers = [
 
 [tool.poetry.dependencies]
 python = ">=3.7.1,<4"
-atpublic = "^2.3"
-cached_property = "^1"
-importlib-metadata = { version = "^4", python = "<3.8" }
-multipledispatch = "^0.6"
-numpy = "^1"
-pandas = "^1.2.5"
-parsy = "^1.3.0"
+atpublic = ">=2.3,<3"
+cached_property = ">=1,<2"
+importlib-metadata = { version = ">=4,<5", python = "<3.8" }
+multipledispatch = ">=0.6,<0.7"
+numpy = ">=1,<2"
+pandas = ">=1.2.5,<2"
+parsy = ">=1.3.0,<2"
 pytz = "^2021.1"
 regex = "^2021.7.6"
-cytoolz = { version = "^0.11", python = "<3.10" }
-toolz = "^0.11"
+cytoolz = { version = ">=0.11,<0.12", python = "<3.10" }
+toolz = ">=0.11,<0.12"
 clickhouse-driver = { version = ">=0.1,<0.3", optional = true }
-clickhouse-sqlalchemy = { version = "^0.1.4", optional = true }
+clickhouse-sqlalchemy = { version = ">=0.1.4,<0.2", optional = true }
 dask = { version = "^2021.2.0", optional = true, extras = [
   "array",
   "dataframe"
 ] }
 geoalchemy2 = { version = ">=0.6,<0.10", optional = true }
 geopandas = { version = ">=0.6,<0.11", optional = true }
-graphviz = { version = "^0.16", optional = true }
-hdfs = { version = "^2", optional = true, extras = ["kerberos"] }
+graphviz = { version = ">=0.16,<0.20", optional = true }
```

Comment:
`<0.17`?

Reply:
nah, this is updating graphviz as well, from the #3157